js-core/homeworks/homework-15/src/main.js

```diff
@@ -0,0 +1,386 @@
+/*
+TASK 0
+Проверьте что строка содержит все символы от "a" до "z"
+  solution("wyyga") // false
+  solution("qwertyuioplkjhgfdsazxcvbnm") // true
+  solution("ejuxggfsts") // false
+  solution("qpwoeirutyalskdjfhgmznxbcv") // true
+  solution("qqqqqqqqpwoeirutyallskkdjfhgmmznxbcv") // true
+  solution("0123456789abcdefghijklmnop") // false
+*/
+
+const solution = str => {
+    const turnStrToArray = str.toLowerCase().split('');
+    const uniqueLetters = new Set();
+    
+    turnStrToArray.forEach(letter => {
+        uniqueLetters.add(letter);
+    });
+    
+    return uniqueLetters.size === 26 ? true : false;
+};
+
+const testStr = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+const testStr2 = 'test';
+
+console.log(solution(testStr));
+console.log(solution(testStr2));
+
+/*
+ 2. Напишите функция которая преобразовывает / открывает
+ скобки всех вложенных внутри массивов
+ Необходимо реализовать рекурсивный фызов функции.
+ Функция должна открывать любое количество
+ внутренних массивов и объектов
+ example:
+ [[1,2],[3,[4]],5, 10] => [1, 2, 3, 4, 5, 10]
+ [25, 10, [10, [15]]] => [25, 10, 10, 15]
+ [1, [2, [ {a: "b", c: 'd' }, { c: [1, 2, 5] } ] ] ] => [1, 2, {a: "b"}]
+ */
+
+//#1 arr == [...] flattenedArray = [1] + flatten = [2, [{a: "b"}, { c: 'd' }]]
+//#2 arr == [2, [ {a: "b"}, { c: 'd' } ] ] flattenedArray = [2] + flatten == [{a: "b"}, { c: 'd' }]
+//#3 arr == [ {a: "b"}, { c: 'd' } ] flattenedArray = [{a: "b"}, { c: 'd' }]
+//#
+const flatten = arr => {
+    return arr.reduce(function(newArray, elem) {
+    return Array.isArray(elem) ? newArray.concat(flatten(elem)) : newArray.concat(elem);
+  }, []);
+};
+
+console.log(flatten([[1,2],[3,[4]],5, 10]));
+console.log(flatten([25, 10, [10, [15]]]));
+console.log(flatten([1, [2, [ {a: "b", c: 'd' }, { c: [1, 2, 5] } ] ] ]));
+
+
+/*
+Виртуализировать таблицу, сделать рендер всей
+таблицы через JavaScript.
+Второй макет.
+https://github.com/aleksandra-maslennikova/telephone-book/blob/master/index.html
+Выглядеть должно так же: https://aleksandra-maslennikova.github.io/telephone-book/index.html
+*/
+
+const contentForThead = ['Name', 'Last Name', 'Email'];
+
+const users = [
```

Comment:
please move all logic relates to PhoneBook to another directory and work on it independently from home tasks